src/Interpreters/Cache/FileSegment.cpp

```diff
@@ -932,8 +951,8 @@ bool FileSegment::assertCorrectnessUnlocked(const FileSegmentGuard::Lock & lock)
             return;
 
         const auto & entry = it->getEntry();
-        if (download_state != State::DOWNLOADING && entry->size != reserved_size)
-            throw_logical(fmt::format("Expected entry.size == reserved_size ({} == {})", entry->size.load(), reserved_size.load()));
+        if (download_state != State::DOWNLOADING && entry->getSize(IFileCachePriority::Entry::SizeAlignment::NOT_ALIGNED) != reserved_size)
```

Comment:
Let's add a comment in `FileSegment.h` near `size_t reserved_size = 0;` and `size_t downloaded_size = 0`-  `/// Contains non-aligned size`